packages/types/src/client.ts

```diff
@@ -147,4 +148,13 @@ export interface Client<O extends ClientOptions = ClientOptions> {
    * @param event The dropped event.
    */
   recordDroppedEvent(reason: EventDropReason, dataCategory: DataCategory, event?: Event): void;
+
+  // HOOKS
+
+  on(
+    hook: 'startTransaction' | 'transactionFinish',
```

Comment:
m: Better use the type from types/src/hook.ts here?

Reply:
Ah, nevermind, we'll overload this. We should just overload `emit` the same way then 👍 
maybe still makes sense to have a function signature that can be re-used between on and emit, though? e.g.:

```ts
type HookTransactionName = 'startTransaction' | 'transactionFinish';
type HookTransactionParams = [Transaction];

on(hook: HookTransactionName, callback: (...HookTransactionParams): void;
emit(hook: HookTransactionName, ...HookTransactionParams): void;
```